core/src/main/java/org/apache/accumulo/core/clientImpl/ThriftScanner.java

```diff
@@ -299,10 +305,52 @@ public ScanState(ClientContext context, TableId tableId, Authorizations authoriz
     }
   }
 
-  public static class ScanTimedOutException extends IOException {
+  static <T> Optional<T> waitUntil(Supplier<Optional<T>> condition, Duration maxWaitTime,
```

Comment:
It does seem like there could be some consolidation with these however I'm not sure what the best way to consolidate them would be.

Reply:
If it were me, I would do the following in `main`:

1. Move `Wait` to `org.apache.accumulo.core.util`
2. Move `UtilWaitThread.sleep` into `Wait`
3. Add this method to `Wait`

Reply:
This method has some things very specific to scans in it so it could not be moved as is.  However it could build on a more generic method and pass the scan specific code in as a lambda.